File: coders_fullstack_code_eval_attempter/python/countdown_illustrator.py
# -*- coding: utf-8 -*-
"""
Created on Wed Dec 18 19:15:50 2024

@author: alima
"""

import time 

def countdown_and_alert(time_remaining): 
    while time_remaining > 0: 
        mins, secs = divmod(time_remaining, 60) 
        timer_display = '{:02d}:{:02d}'.format(mins, secs) 
        print(f"Time remaining: {timer_display}", end="\r") 
        time.sleep(1) 
        time_remaining -= 1
    
    print('Boom! Time\'s up!!') 

try:
    user_input = int(input("Set the countdown timer (in seconds): "))  
    countdown_and_alert(user_input)
except ValueError:
    print("Invalid input! Please enter a valid integer value for the countdown timer.")

File: coders_fullstack_code_eval_attempter/C++/count_down_illustrator.cpp
#include <iostream>
#include <iomanip>   // For std::setw and std::setfill
#include <chrono>    // For std::chrono::seconds
#include <thread>    // For std::this_thread::sleep_for
#include <string>    // For std::string
#include <stdexcept> // For std::invalid_argument and std::out_of_range exceptions

void countdown_and_alert(int time_remaining) {
    while (time_remaining > 0) {
        int mins = time_remaining / 60;
        int secs = time_remaining % 60;
        std::cout << "Time remaining: "
                  << std::setw(2) << std::setfill('0') << mins << ":"
                  << std::setw(2) << std::setfill('0') << secs
                  << "\r" << std::flush;
        std::this_thread::sleep_for(std::chrono::seconds(1));
        --time_remaining;
    }
    std::cout << std::endl << "Boom! Time's up!!" << std::endl;
}

int main() {
    std::cout << "Set the countdown timer (in seconds): ";
    std::string user_input_str;
    std::getline(std::cin, user_input_str);

    try {
        int user_input = std::stoi(user_input_str);
        if (user_input < 0) {
            throw std::invalid_argument("Negative value");
        }
        countdown_and_alert(user_input);
    } catch (const std::invalid_argument&) {
        std::cout << "Invalid input! Please enter a valid integer value for the countdown timer." << std::endl;
    } catch (const std::out_of_range&) {
        std::cout << "Input value is out of range." << std::endl;
    }

    return 0;
}